source-code/sdk-js/src/adapter-sveltekit/vite-plugin/transforms/refactoring.md

```diff
@@ -209,27 +427,181 @@ JavaScript files can run on the server and on the client. So we need to detect t
+   4. wrap markup (the reactive $ will be added by `transformSvelte`)
+      1. if `config.languageInUrl && config.isStatic` with `{#if $$_INLANG_LANGUAGE_$$}{/if}`
+      2. if `config.languageInUrl` with `{#key $$_INLANG_LANGUAGE_$$}{/key}`, else `{#if $$_INLANG_LANGUAGE_$$}{/if}`
+
+2. Pass everything to `transformSvelte`
+
+Possible implementation:
+
+```js
+// NOTES @benjaminpreiss I am not sure whether we can pull `module` out of the preprocessing api
+function transformLayoutSvelte({ markup, style, script }, config) {
+	// Import all necessary things
+	if (config.isRoot) {
+		imports(script.ast, "@inlang/sdk-js").add("language")
+		imports(script.ast, "@inlang/sdk-js/adapter-sveltekit/shared").add("getRuntimeFromData")
+		imports(
+			script.ast,
+			`@inlang/sdk-js/adapter-sveltekit/client/${config.languageInUrl ? "not-" : ""}reactive`,
+		).add("addRuntimeToContext")
+		if (!config.languageInUrl)
+			imports(script.ast, "@inlang/sdk-js/adapter-sveltekit/client/reactive").add("localStorageKey")
+
+		// Insert reactive inlang sdk import reassigments
+		if (!config.languageInUrl)
+			contexts(script.ast).insertAfterImports(
+				parse(`
+          $: {
+            addRuntimeToContext(getRuntimeFromData(data));
+            $$_INLANG_REASSIGN_SDK_IMPORTS_$$;
+          }
+        `),
+			)
+		// Insert `export let data` and `addRuntimeToContext(getRuntimeFromData(data));`
+		// As you can see, contexts() with only one param returns the top level context
+		contexts(script.ast).insertAfterImports(
+			parse(`
+        export let data;
+        addRuntimeToContext(getRuntimeFromData(data));
+      `),
+		)
+		// Insert slot, only if markup is empty
+		if (isEmpty(script.markup)) contexts(markup).insert(`<slot/>`)
+		// Insert a conditional
+		if (config.languageInUrl && config.isStatic)
+			contexts(markup).wrap(`{#if $$_INLANG_LANGUAGE_$$}`, `{/if}`)
+		contexts(markup).wrap(
+			`{#${config.languageInUrl ? "key" : "if"} $$_INLANG_LANGUAGE_$$}`,
+			`{/${config.languageInUrl ? "key" : "if"}}`,
+		)
+	}
+	return transformSvelte({ markup, style, script }, config)
+}
+```
 
 ### `+page.svelte`
 
 - pass everything to `*.svelte`
 
+Possible implementation:
+
+```ts
+function transformPageSvelte(
+	{
+		markup,
+		style,
+		script,
+	}: { markup: { ast: Node; magicString: MagicString }; style: Node; script: { ast: Node } },
+	config,
+) {
+	return transformSvelte({ markup, style, script }, config)
+}
+```
+
 ### `*.svelte`
 
 1.  replace `@inlang/sdk-js` imports with `getRuntimeFromContext`
-2.  resolve aliases and placeholders
-3.  if reactive
+2.  Add import:
+
+    - if `config.languageInUrl`: `import { getRuntimeFromContext } from "@inlang/sdk-js/adapter-sveltekit/client/reactive"`
+    - else: `import { getRuntimeFromContext } from "@inlang/sdk-js/adapter-sveltekit/client/reactive"`
+
+3.  resolve aliases and placeholders `// NOTES @ivanhofer - what do you mean with this step?`
```

Comment:
`$$_INLANG_I_$$` => `i`
and
`import { i as inl } from '@inlang/sdk-js'`
`$$_INLANG_I_$$('test')` => `inl('test')`